eclipse-collections-code-generator/src/main/resources/api/primitiveIterable.stg

```diff
@@ -127,6 +127,96 @@ public interface <name>Iterable extends PrimitiveIterable
         }
     }
 
+    /**
+     * Returns true if any of the values specified in the source array are contained
+     * in the <name>Iterable, and false if they are not.
+     *
+     * @since 11.0
+     */
+    default boolean containsAny(<type>... source)
+    {
+        <name>Iterable inside = this;
+        if (this.size() > 32 && source.length > 32 && !(this instanceof <name>Set))
+        {
+            inside = this.toSet();
+        }
+        for (<type> item : source)
+        {
+            if (inside.contains(item))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if any of the values specified in the source <name>Iterable are contained
+     * in the <name>Iterable, and false if they are not.
+     *
+     * @since 11.0
+     */
+    default boolean containsAny(<name>Iterable source)
+    {
+        <name>Iterable outside = this;
+        <name>Iterable inside = source;
+        if (this.size() \< source.size())
+        {
+            outside = source;
+            inside = this;
+        }
+        if (outside.size() > 32 && inside.size() > 32 && !(inside instanceof <name>Set))
```

Comment:
at this point `inside` is guaranteed to be the smaller of the two. Therefore, `inside.size() > 32` implies `outside.size() > 32`. We can drop `outside.size() > 32`. Same for `noneSatisfy`

Reply:
Updated. 😄